src/prefect/server/api/flow_runs.py

```diff
@@ -520,27 +520,142 @@ async def set_flow_run_state(
     return orchestration_result
 
 
+async def _handle_paused_flow_run_input(
```

Comment:
This looks like the same handling that `set_state` does. Would it make sense to pull the validation stuff out into its own function and then use this in both places? Seems like a generally useful thing for a function to do.

Reply:
I considered it! There were some logic differences at first, and I'm still not convinced it should work the way it does in this PR, where the incoming key must match the pause input key of a paused flow run for us to resume it. So we might change that after releasing. Also, it's mostly repetitive error handling, but the actual error handling is different between the two, so we'd get an explosion of repetition by abstracting... I'd like wait and see if that's truly necessary.

Reply:
Wait, @bunchesofdonald, did you mean `resume_flow_run`? I thought that's what you were talking about.